javascript/javascript1/week2/candyHelper.js

```diff
@@ -0,0 +1,35 @@
+const boughtCandyPrices = [];
+const candyPrices = {
+  sweet: 0.5,
+  chocolate: 0.7,
+  toffee: 1.1,
+  chewingGum: 0.03,
+};
+
+function addCandy(candyType, weight) {
+  const pricePerGram = candyPrices[candyType];
+  if (pricePerGram) {
+    const price = pricePerGram * weight;
+    boughtCandyPrices.push(price);
+  } else {
+    console.log("Candy type not recognized.");
+  }
+}
+
+function canBuyMoreCandy() {
+  const amountToSpend = Math.random() * 100;
```

Comment:
`amountToSpend` is being generated inside canBuyMoreCandy(), meaning it changes every time the function is called. This causes inconsistent behavior, since the amount to spend should stay the same.